src/LandIce/problems/LandIce_StokesFOThermoCoupled.hpp

```diff
@@ -303,9 +302,7 @@ constructEnthalpyEvaluators (PHX::FieldManager<PHAL::AlbanyTraits>& fm0,
   //Output
   p->set<std::string>("Basal Vertical Velocity Variable Name", "basal_vert_velocity_" + basalSideName);
   p->set<std::string>("Basal Melt Rate Variable Name", "basal_melt_rate_" + basalSideName);
-  ev = createEvaluatorWithTwoScalarTypes<BasalMeltRate,EvalT>(p,dl->side_layouts[basalSideName],
-                                                                         FST::Scalar,
-                                                                         get_scalar_type(melting_enthalpy_name));
+  ev = createEvaluatorWithOneScalarType<BasalMeltRate,EvalT>(p,dl->side_layouts[basalSideName], FST::Scalar);
```

Comment:
Since the scalar type is known at compile time, you might be better off doing
```
ev = Teuchos::rcp(new BasalMeltRate<EvalT, PHAL::AlbanyTraits, typename EvalT::ScalarT>(*p,dl);
```
like you do below for LiquidWaterFraction. It should be faster to compile.

Reply:
It makes sense, will do.